text/0018-publish-update.md

```diff
@@ -0,0 +1,180 @@
+
+Speaking of updates, I don't have any clear mechanism for re-fetching datasets that's been updated since I "added" it. We need a quick and easy way to syncronize my remote copy of a dataset with the lastest version after it's been published.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This RFC introduces two new concepts to Qri that will come with new commands and, will modify some existing behaviors. Publish and Update will work together to allow users to explicitly opt-in to sync points
+
+Because this process will be run frequently, I'd like to make the common use patterns as simple as possible, to the point that `qri save` should come with a `publish` flag that combines saving an update & publishing it into one step.
+
+## Publish
+All datasets start as "unlisted", and the user must `publish` them to make visible to other users. Peers calling `list_datasets` now only see datasets you've published, at version that's been most-recently published. If I create a dataset, then publish, and save changes to that dataset, I must re-publish the new changes for others to see the update.
+
+Whenever I choose, I may "unpublish" a dataset, which will remove it entirely from the list returned by `list_datasets`. It is true that others may still have a copy of my dataset. At present there's not much I can do about it.
+
+#### Publish will integrate with Registries
+Running `publish` with a configured registry should POST a dataset _summary_ to the registry. We haven't defined dataset previews, summaries, and documents yet (a story for another RFC), but once it's done publish will both list the dataset for p2p discovery and push the dataset to the registry.
+
+## Update
+Update brings your data to the latest known version of a dataset. Update comes in two flavors:
+* Updating Someone else's dataset
+* Updating A Dataset You Own
+Both types of update will be accessed via the same command like and API interfaces, they only differ based on the dataset reference in question.
+
+#### Updating a peer's dataset
+Running update on a peer dataset checks to see if the peer is online and calls `get_dataset`, if the version is different, it'll grab that dataset. If the peer in question is offline we can't check for updates. Once publish pushes to the registry we can check the registry for updates.
+
+Later on we'll want to do proactive checks for updates by submitting a list of dataset refs we'd like to grab updates for to peers and registries.
+
+#### Updating a dataset you own
+running update on a dataset in your namespace will check if a transform script is specified, If so, it'll re-execute the transform & check for changes. If changes exist it'll create a commit.
+
+Manual changes to a dataset are still run with `qri save`. Calling update on a dataset with no transform script is an error. That error should tell users to run `qri save`.
```

Comment:
If I manually change something using `qri save` does that remove the old transform script? Or if not, and it keeps the transform script, is that script re-run each time?

Reply:
manual changes remove a transform script by setting `transform: null`, otherwise the transform will remain. On both save and update transforms are always run. More generally, existing transforms always run.

It would be possible to hack the transform to only run in specific conditions with logic that reads a config or secret that's fed in each time, so maybe we should consider a `--no-transform` flag just to formalize it 🤷‍♂️? I'd rather wait to see if people really need the distinction.